### PR TITLE
DB-7626: Add Spark Adapter API to split table based on sampling a custom DataFrame

### PR DESCRIPTION
**Pull request by jyuanca**

### CODE REVIEW

**Comment by yxia92** [comment]
@jyuanca The subject says add spark adapter API, but I didn't find the API change  in SplicemachineContext.scala, did I miss something? Thanks.

**Comment by jyuanca** [comment]
@yxia92 sorry, I forgot to push some changes.

**Comment by yxia92** [comment]
Got it, I see the relevant code in the 2.5 PR. One more question, where is the code that does the shuffling before sampling? Or is it part of dataset.sampleWithoutReplacement()?

**Comment by jyuanca** [comment]
@yxia92 Shuffle is not addressed in this PR.

File: splice_spark/src/main/spark2.1/com/splicemachine/spark/splicemachine/SplicemachineContext.scala [hunk 1]
```diff
@@ -325,6 +325,28 @@ class SplicemachineContext(options: Map[String, String]) extends Serializable {
...
+    *
+    * Sample the dataframe, split the table, and insert a dataFrame into a table (schema.table).  This corresponds to an
+    *
+    * insert into from select statement
+    *
+    * @param dataFrame
+    * @param schemaTableName
+    * @param sampleFraction
+    */
+  def insert(dataFrame: DataFrame, schemaTableName: String, sampleFraction: Double): Unit = {
```
Comment (dgomezferro) [inline splice_spark/src/main/spark2.1/com/splicemachine/spark/splicemachine/SplicemachineContext.scala]: I would name this method something more explicit: splitAndInsert, insertAfterSplitting, insertWithSplits ...

Comment (jyuanca) [reply splice_spark/src/main/spark2.1/com/splicemachine/spark/splicemachine/SplicemachineContext.scala]: API was changed to splitAndInsert

**Pull request merged**